internal/plugin/builtin/pluginapigen/handle.go

```diff
@@ -0,0 +1,256 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// Package pluginapigen provides a plugin Handle that generates code used by the
+// plugin system itself.
+//
+// This is made available as a hidden flag: "--generate-plugin-api".
+package pluginapigen
+
+import (
+	"path/filepath"
+	"strings"
+
+	iplugin "github.com/thriftrw/thriftrw-go/internal/plugin"
+	"github.com/thriftrw/thriftrw-go/plugin"
+	"github.com/thriftrw/thriftrw-go/plugin/api"
+)
+
+// Handle is a plugin.Handle that generates code for the plugin system API.
+var Handle iplugin.Handle = handle{}
```

Comment:
Suggest `intplugin` instead of `iplugin` - feels like an Apple product announcement otherwise.